payment/order.go

```diff
@@ -150,3 +186,93 @@ func CreateOrderItemFromMacaroon(sku string, quantity int) (*OrderItem, error) {
 func (order Order) IsPaid() bool {
 	return order.Status == "paid"
 }
+
+// IsStripePayable returns true if every item is payable by Stripe
+func (order Order) IsStripePayable() bool {
+	for _, item := range order.Items {
+
+		// check stripe in payment
+		if !strings.Contains(item.PaymentMethods, STRIPE_PAYMENT_METHOD) {
+			return false
+		}
+		// TODO: make sure we have a stripe_product_id caveat
+		// TODO: if not we need to look into subscription trials:
+		/// -> https://stripe.com/docs/billing/subscriptions/trials
+
+	}
+	return true
+}
+
+type CreateCheckoutSessionResponse struct {
+	SessionID string `json:"checkoutSessionId"`
+}
+
+// Create a Stripe Checkout Session for an Order
+func (order Order) CreateStripeCheckoutSession(email string) CreateCheckoutSessionResponse {
+	// os.Getenv("STRIPE_SECRET")
+	stripe.Key = "sk_test_51HlmudHof20bphG6m8eJi9BvbPMLkMX4HPqLIiHmjdKAX21oJeO3S6izMrYTmiJm3NORBzUK1oM8STqClDRT3xQ700vyUyabNo"
+
+	// Create customer if not already created
+	i := customer.List(&stripe.CustomerListParams{
+		Email: stripe.String(email),
+	})
+
+	matchingCustomers := 0
+	for i.Next() {
+		matchingCustomers++
+	}
+
+	var customerID string
+	if matchingCustomers > 0 {
+		customerID = i.Customer().ID
+	} else {
+		customer, _ := customer.New(&stripe.CustomerParams{
+			Email: stripe.String(email),
+		})
+		customerID = customer.ID
+	}
+
+	// os.Getenv("STRIPE_SUCCESS_URL")
+	successUrl := stripe.String("https://together.bsg.brave.software/")
```

Comment:
this should probably be a parameter passed during order creation with some validation ( e.g. must be considered samesite as the location of all items in the order )